### ISSUE TITLE
TestAbort_ERC20_XMRMakerCancels flaky

### ISSUE THREAD

**Issue by dimalinux** [AthanorLabs/atomic-swap#332]
`TestAbort_ERC20_XMRMakerCancels` is the first integration test to run. Its ETH counterpart test, `TestAbort_XMRTakerCancels`, runs much later and always succeeds. We've tried updating the pause in the script that starts the `swapd` daemons to 30 seconds and the test is still failing regularly. The original thought was that `TestAbort_XMRTakerCancels` succeeds, because the daemons have been running for a considerably longer period of time, but at 30 seconds, our network startup code is buggy if there is a timing issue on a localhost only test.
 ```
=== RUN   TestRunIntegrationTests/TestAbort_ERC20_XMRMakerCancels
    integration_test.go:667: 
        	Error Trace:	/home/runner/work/atomic-swap/atomic-swap/tests/integration_test.go:667
        	            				/home/runner/work/atomic-swap/atomic-swap/tests/erc20_integration_test.go:62
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRunIntegrationTests/TestAbort_ERC20_XMRMakerCancels
        	Messages:   	peer count mismatch
```
The failing lines are:
```
	peerIDs, err := c.Discover(string(coins.ProvidesXMR), defaultDiscoverTimeout)
	require.NoError(s.T(), err)
	require.Equalf(s.T(), 1, len(peerIDs), "peer count mismatch")
```
So we are semi-frequently failing to discover the peer with an offer when the asset is a token, but not when the asset is ETH.

**Comment by dimalinux** [comment]
I ruled out the possibility that the test is is related to the asset not being ETH.

I renamed `TestAbort_XMRTakerCancels` to `TestAbort_AAA_XMRTakerCancels` so it runs first. It will fail in the same place where `TestAbort_ERC20_XMRMakerCancels` frequently fails, then `TestAbort_ERC20_XMRMakerCancels` will run 2nd and pass.

**Comment by dimalinux** [comment]
Here's what I think is happening:

When an offer is made, we are queuing the advertisement that the node offers XMR. The advertisement does not go out immediately.  If the last loop ran 1 seconds ago, we'll need to make sure we wait over 29 seconds to know that the advertisement went out. I think we need to change the code so that the initial node advertisement for XMR goes out immediately without being queued. Maybe we decouple the loop for "" from the loop for XMR?

**Comment by dimalinux** [comment]
When fixing this issue, should we continue to advertise `XMR` for any offer?  Another option would be to advertise `XMR-ETH` for a provider of XMR in exchange for ETH and `XMR-{TOKEN-NAME-OR-ADDR}` for a provider XMR in exchange for some token.

**Comment by dimalinux** [comment]
Closing.  Fixes were made in #333 and #334.
